### PR TITLE
Fix prysmctl writing empty JSON/YAML files

### LINKED ISSUE #12576: "prysmtcl testnet generate-genesis" command does not output genesis state in json format
# 🐞 Bug Report

### Description

I am setting up a devnet using geth and prysm beacon chain. I need to output the generated genesis state for the beacon chain to a JSON file to do some manipulation. The console logs say the command was completed successfully but an empty object is written to the JSON file. Using the `--output-ssz` flag outputs the state fine and it's able to run the network. 

Command:
`./prysmctl testnet generate-genesis --fork bellatrix --num-validators 8 --output-json ./test-genesis.json --chain-config-file=./consensus/config.yml --geth-genesis-json-in ./execution/genesis.json --geth-genesis-json-out ./execution/genesis.json`

Output:
```
INFO[0000] Specified a chain config file: ./consensus/config.yml  prefix=genesis
INFO[0000] No genesis time specified, defaulting to now()  prefix=genesis
INFO[0000] setting fork geth times                       prefix=genesis shanghai=0xc000776a30
INFO[0000] Done writing genesis state to ./test-genesis.json  prefix=genesis
INFO[0000] Command completed                             prefix=genesis
```

### Has this worked before in a previous version?

Unknown

## 🔬 Minimal Reproduction

`./prysmctl testnet generate-genesis --num-validators 8 --output-json ./test-genesis.json`

## 🔥 Error

N/A

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
macOS Version 13.4
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
v4.0.6
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

N/A

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

This PR makes it so that we can use JSON and YAML Marshal on `state.BeaconState`. 
Thus allowing `prysmctl` to write the genesis data to a file.

**Which issues(s) does this PR fix?**

Fixes #12576

**Other notes for review**

### CODE REVIEW

File: beacon-chain/state/state-native/beacon_state_mainnet.go [hunk 1]
```diff
@@ -13,6 +14,7 @@ import (
 	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
 	enginev1 "github.com/prysmaticlabs/prysm/v4/proto/engine/v1"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"gopkg.in/yaml.v2"
```
Comment (james-prysm) [inline beacon-chain/state/state-native/beacon_state_mainnet.go:17]: you should use the kubernetes package instead of this one

Comment (james-prysm) [reply beacon-chain/state/state-native/beacon_state_mainnet.go:17]: you also don't need yaml: in the attribute only json

File: beacon-chain/state/state-native/beacon_state_mainnet.go [hunk 2]
```diff
@@ -60,3 +62,114 @@ type BeaconState struct {
...
+		CurrentEpochParticipation:           b.currentEpochParticipation,
+		JustificationBits:                   b.justificationBits,
+		PreviousJustifiedCheckpoint:         b.previousJustifiedCheckpoint,
+		CurrentJustifiedCheckpoint:          b.currentJustifiedCheckpoint,
+		FinalizedCheckpoint:                 b.finalizedCheckpoint,
+		InactivityScores:                    b.inactivityScores,
+		CurrentSyncCommittee:                b.currentSyncCommittee,
+		NextSyncCommittee:                   b.nextSyncCommittee,
+		LatestExecutionPayloadHeader:        b.latestExecutionPayloadHeader,
+		LatestExecutionPayloadHeaderCapella: b.latestExecutionPayloadHeaderCapella,
```
Comment (james-prysm) [inline beacon-chain/state/state-native/beacon_state_mainnet.go:132]: you will probably need to do this by fork or something see this is only for capella and will need to be updated for deneb. is there a way you can use the existing datastructure elsewhere?

File: beacon-chain/state/state-native/beacon_state_mainnet.go [hunk 3]
```diff
@@ -60,3 +62,114 @@ type BeaconState struct {
 	merkleLayers          [][][]byte
 	sharedFieldReferences map[types.FieldIndex]*stateutil.Reference
 }
+
+type beaconStateMarshalable struct {
+	Version                             int                                     `json:"version" yaml:"version"`
+	GenesisTime                         uint64                                  `json:"genesis_time" yaml:"genesisTime"`
+	GenesisValidatorsRoot               [32]byte                                `json:"genesis_validators_root" yaml:"genesisValidatorsRoot"`
```
Comment (rauljordan) [inline beacon-chain/state/state-native/beacon_state_mainnet.go:69]: snake case yaml too

File: beacon-chain/state/state-native/beacon_state_mainnet.go (hunk 2 above)
Comment (james-prysm) [reply beacon-chain/state/state-native/beacon_state_mainnet.go:131]: I think this object might live somewhere else as well too 🤔

**Review by rauljordan — APPROVED** [review]

**Pull request merged**